File: webapi/Models/HR/PersonalData.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace webapi.Models.HR
{
    public record PersonalData
    {
        [Key]
        public int Id { get; set; }
        public required string IdentityText { get; set; }
        public int IdentityCode { get; set; }
        public Nullable<DateOnly> BirthDate { get; set; }
        public Nullable<Gender> Gender { get; set; }
        public string? PersonalIdNumber { get; set; }
        public Nullable<DateOnly> PersonalIdIssueDate { get; set; }
        public string? PersonalIdIssueBy { get; set; }
        public Nullable<int> AddressId { get; set; }
        public Address? Address { get; set; }
    }

    public record PersonalDataDTO
    {
        public required string IdentityText { get; set; }
        public int IdentityCode { get; set; }
        public Nullable<DateOnly> BirthDate { get; set; }
        public Nullable<Gender> Gender { get; set; }
        public string? PersonalIdNumber { get; set; }
        public Nullable<DateOnly> PersonalIdIssueDate { get; set; }
        public string? PersonalIdIssueBy { get; set; }
        public Nullable<int> AddressId { get; set; }
    }
}

File: reactapp/src/pages/Employees/models/PersonalData.ts
import { Optional } from "@/types/Optional";
import Gender from "@app-types/enums/Gender";

export interface PersonalData {
    id: int
    personalEmail: string,
    workEmail: string,
    identityText: string,
    identityCode: int,
    birthDate: Optional<string>,
    gender: Optional<Gender>,
    personalIdNumber: Optional<string>,
    personalIdIssueDate: Optional<string>,
    personalIdIssueBy: Optional<string>,
    addressId: Optional<int>
}

export const defaultPDataUpdateDTO: PersonalData = {
    id: 0,
    personalEmail: "",
    workEmail: "",
    identityText: "",
    identityCode: 0,
    birthDate: null,
    gender: null,
    personalIdNumber: null,
    personalIdIssueDate: null,
    personalIdIssueBy: null,
    addressId: null
}

export interface PersonalDataDTO {
    personalEmail: string,
    workEmail: string,
    identityText: string,
    identityCode: int,
    birthDate: Optional<string>,
    gender: Optional<Gender>,
    personalIdNumber: Optional<string>,
    personalIdIssueDate: Optional<string>,
    personalIdIssueBy: Optional<string>,
    addressId: Optional<int>
}


export const defaultPersonalDataDTO: PersonalDataDTO = {
    personalEmail: "",
    workEmail: "",
    identityText: "",
    identityCode: 0,
    birthDate: null,
    gender: null,
    personalIdNumber: null,
    personalIdIssueDate: null,
    personalIdIssueBy: null,
    addressId: null
}